client_cert_test.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"crypto/tls"
+	"errors"
+	"testing"
+)
+
+func TestReadClientCertNoFilePaths(t *testing.T) {
+	if _, err := readClientCert("certPath", ""); err != nil {
```

Comment:
These tests still require some work. Get rid of mocks and global _var_, add some valid and invalid _cert_ and _key_ files to _git_ to test against them.